AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs

```diff
@@ -18,18 +18,82 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+using AstarteDeviceSDK.Protocol;
 using AstarteDeviceSDKCSharp.Data;
+using AstarteDeviceSDKCSharp.Protocol.AstarteEvents;
 
 namespace AstarteDeviceSDKCSharp.Protocol
 {
-    public class AstarteServerPropertyInterface : AstartePropertyInterface
+    public class AstarteServerPropertyInterface : AstartePropertyInterface,
+    IAstarteServerValueBuilder, IAstarteServerValuePublisher
     {
         private readonly IAstartePropertyStorage _propertyStorage;
+        private readonly List<IAstartePropertyEventListener> _listeners;
 
-        public AstarteServerPropertyInterface(IAstartePropertyStorage propertyStorage) : base(propertyStorage)
+        public AstarteServerPropertyInterface(IAstartePropertyStorage propertyStorage)
+        : base(propertyStorage)
         {
             _propertyStorage = propertyStorage;
+            _listeners = new();
         }
 
+        public AstarteServerValue? Build(string interfacePath, object? serverValue,
+        DateTime timestamp)
+        {
+            AstarteInterfaceMapping? targetMapping = null;
+            AstarteServerValue? astarteServerValue;
+
+            foreach (var entry in GetMappings())
+            {
+                if (AstarteInterface.IsPathCompatibleWithMapping(interfacePath, entry.Key))
+                {
+                    targetMapping = entry.Value;
+                    break;
+                }
+            }
+
+            if (targetMapping != null)
+            {
+                object? astarteValue = serverValue;
+                if (targetMapping.GetType() == typeof(DateTime))
+                {
+                    astarteValue = Convert.ToDateTime(serverValue);
+                }
+
+                astarteServerValue =
+                        new AstarteServerValue.AstarteServerValueBuilder(astarteValue)
+                        .InterfacePath(interfacePath)
+                        .Build();
+            }
+            else
+            {
+                astarteServerValue = null;
+            }
+
+            return astarteServerValue;
+        }
```

Comment:
I think you can move the `tagetMapping` block into the foreach and have a more compact block. 
```suggestion
    AstarteServerValue? astarteServerValue = null;

    foreach (var entry in GetMappings())
    {
        if (AstarteInterface.IsPathCompatibleWithMapping(interfacePath, entry.Key))
        {
            AstarteInterfaceMapping targetMapping = entry.Value;
            object? astarteValue = serverValue;
            if (targetMapping.GetType() == typeof(DateTime))
            {
               astarteValue = Convert.ToDateTime(serverValue);
            }

            astarteServerValue =
                    new AstarteServerValue.AstarteServerValueBuilder(astarteValue)
                   .InterfacePath(interfacePath)
                   .Build();
              break;           
        }
    }    
    return astarteServerValue;
 }
```

Reply:
I made changes.